src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs

```diff
@@ -811,5 +819,66 @@ public async Task CreateGoodBreakpointAndHitGoToWasmPageWithoutAssetsComeBackAnd
                 }
             );
         }
+
+        [Fact]
+        public async Task BreakInDefaultInterfaceMethod()
+        {
+            string methodName = "DefaultMethod";
+            string assembly = "dotnet://debugger-test.dll/debugger-test.cs";
+            var bp_dim = await SetBreakpointInMethod("debugger-test.dll", "IDefaultInterface", methodName, 1);
+            var init_location = await EvaluateAndCheck(
+                $"window.setTimeout(function() {{ invoke_static_method('[debugger-test] DefaultInterfaceMethod:Evaluate'); }}, 1);",
+                assembly,
+                bp_dim.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_dim.Value["locations"][0]["columnNumber"].Value<int>(),
+                methodName,
+                wait_for_event_fn: async (pause_location) => { await CheckDefaultMethod(pause_location); }
+            );
+            await SendCommandAndCheck(null, "Debugger.resume",
+                assembly,
+                bp_dim.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_dim.Value["locations"][0]["columnNumber"].Value<int>(),
+                methodName,
+                wait_for_event_fn: async (pause_location) => { await CheckDefaultMethod(pause_location); }
+            );
+
+            async Task CheckDefaultMethod(JObject pause_location)
+            {
+                Assert.Equal("other", pause_location["reason"]?.Value<string>());
+                Assert.Equal(bp_dim.Value["breakpointId"]?.ToString(), pause_location["hitBreakpoints"]?[0]?.Value<string>());
+
+                var top_frame = pause_location["callFrames"][0];
+                Assert.Equal(methodName, top_frame["functionName"].Value<string>());
+                Assert.Contains("debugger-test.cs", top_frame["url"].Value<string>());
+
+                CheckLocation(assembly, 999, 4, scripts, top_frame["functionLocation"]);
+
+                var scope = top_frame["scopeChain"][0];
+                Assert.Equal("local", scope["type"]);
+                Assert.Equal(methodName, scope["name"]);
+
+                Assert.Equal("object", scope["object"]["type"]);
+                CheckLocation(assembly, 999, 4, scripts, scope["startLocation"]);
+                CheckLocation(assembly, 1002, 4, scripts, scope["endLocation"]);
+                await Task.CompletedTask;
+            }
+        }
+
+        [Fact]
+        public async Task CheckDefaultInterfaceMethodHiddenAttribute()
+        {
+            string methodName = "Evaluate";
+            string assembly = "dotnet://debugger-test.dll/debugger-test.cs";
+            var hidden_bp = await SetBreakpointInMethod("debugger-test.dll", "IOverrideDefaultInterface", "HiddenDefaultMethod", 1);
+            var final_bp = await SetBreakpointInMethod("debugger-test.dll", "DefaultInterfaceMethod", methodName, 7);
+            // check if bp in hidden DIM was ignored:
+            await EvaluateAndCheck(
```

Comment:
This method should confirm that when we stop at `final_bp` the `HiddenDefaultMethod` has already been called. Otherwise this test would pass if `final_bp` was mistakenly put before `HiddenDefaultMethod` was cal

Reply:
Sure, so I will add an assert to check that `scopeChain[0].endLocation.lineNumber == finalBp.lineNumber`. It should be enough to confirm that finalBp is set after we've called `HiddenMethod`.

Reply:
This check depends on `HiddenMethod` being called by `Evaluate` before we hit `final_bp`. If somebody changed `Evaluate` a bit, such that `HiddenMethod` is *not* called before `final_bp` is hit, then this test would pass, and would not be testing what it wants to.

So, I suggest maybe explicitly calling a method that in turn calls `HiddenMethod`? Or something like that to confirm.